### PR TITLE
Application Connector Docker-in-Docker pipelines

### PR DESCRIPTION
**Pull request by mvshao**
**Description**

Changes proposed in this pull request:

- migrate Application Connector k3d-related jobs to DID architecture
- reorganize the structure of the Prow Jobs owned by the Framefrog team

**Related issue(s)**
#7862

### CODE REVIEW

**Review by akgalwas — APPROVED** [review]

File: prow/jobs/kyma-project/kyma/tests/components/application-connector/central-application-connectivity-validator/tests-validator.yaml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        owner: "framefrog"
+      labels:
+        prow.k8s.io/pubsub.project: "sap-kyma-prow"
+        prow.k8s.io/pubsub.runID: "pre-main-kyma-integration-k3d-app-conn-validator"
+        prow.k8s.io/pubsub.topic: "prowjobs"
+        preset-build-pr: "true"
+        preset-gc-project-env: "true"
+        preset-kyma-guard-bot-github-token: "true"
+        preset-kyma-integration-app-connector-tests-app-validator: "true"
+        preset-kyma-integration-central-app-connectivity-enabled: "true"
```
Comment (Ressetkk) [inline prow/jobs/kyma-project/kyma/tests/components/application-connector/central-application-connectivity-validator/tests-validator.yaml:15]: Do you rely on those presets in `k3d-validator-tests` make target?

File: prow/jobs/kyma-project/kyma/tests/components/application-connector/central-application-connectivity-validator/tests-validator.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+              - "bash"
+              - "-c"
+              - "make -C tests/components/application-connector/hack/ci k3d-validator-tests"
+            resources:
+              requests:
+                memory: 1Gi
+                cpu: 500m
+              limits:
+                cpu: 1
+                memory: 4Gi
```
Comment (Ressetkk) [inline prow/jobs/kyma-project/kyma/tests/components/application-connector/central-application-connectivity-validator/tests-validator.yaml:113]: When you run a presubmit job, Prow automatically fetches latest revision of trunk branch (in this case main) and rebases your changes introduced in a PR. This way you will always test the revision you are going to have after merge. I see no sense in running postsubmit which is doing exactly the same thing as presubmit.

File: prow/jobs/kyma-project/kyma/tests/components/application-connector/central-application-connectivity-validator/tests-validator.yaml (hunk 1 above)
Comment (mvshao) [reply prow/jobs/kyma-project/kyma/tests/components/application-connector/central-application-connectivity-validator/tests-validator.yaml:15]: Neither target depends on these preps, I will remove them from any Prow Job.

**Review by Ressetkk — CHANGES_REQUESTED** [review]
I pointed out the part but some remarks about k3d jobs apply also to the ones I didn't comment out. Feel free to catch me on Slack anytime

**Review by Ressetkk — APPROVED** [review]

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**